### ISSUE TITLE
disable "unfocusing steps" after clear() in SelenideElement#setValue()

### ISSUE THREAD

**Issue by yashaka** [selenide/selenide#960]
GIVEN An application with an input field with the submit handler bound to "blur event" (example: edit field `$(".editing .edit")` of a task in any app from todomvc.com). 
WHEN `$(".editing .edit").setValue("new task name")`
THEN: the input field will be:
* old selenium behaviour (at least till ~ 3.6.0)
  **=>** the input field will be submitted with `"new task name"` value
* new selenium webdriver behavior
  **=>** the input field will be submitted with `""` value and then fail with being unable to set the value to "new task name". 

The reason for that is that previously selenium did not do any "events firing" after clear():
```java
    /*
     * ...
     * Note that the events fired by this event may not be as you'd expect.  In particular, we don't
     * fire any keyboard or mouse events.  If you want to ensure keyboard events are fired, consider
     * using something like {@link #sendKeys(CharSequence...)} with the backspace key.  To ensure
     * you get a change event, consider following with a call to {@link #sendKeys(CharSequence...)}
     * with the tab key.
     */
    void clear();
```
but now it [does according to the w3c standard](https://github.com/SeleniumHQ/selenium/commit/e055afda143807049eee5fc4037452e872f07c76#diff-e211904e58bdf54a158dd988160fac38R79): 
![image](https://user-images.githubusercontent.com/1961118/62208888-60168400-b398-11e9-9ea2-ab089c27fc09.png)
It runs "unfocusing events"... 

Somebody does not like this behavior in selenium and considers it [as a bug](https://github.com/w3c/webdriver/issues/1354)

Somebody really [wants this and wants even more events to be fired after clear](https://github.com/w3c/webdriver/issues/1247).

I believe that from the "real user" perspective the new behavior of the `clear` method in selenium - is ok and really natural. If we consider "clear()" as "atomic user operation" on element - then this is natural to fire corresponding events after it. 

But then... if we continue to consider operations from the user perspective, i.e. as if they are "atomic user operations", then now `SelenideElement#setValue` is no atomic any more, since it is based on `clear()` being called before `sendKeys()`... Continuing this logic, I believe that we should find a way to disable "firing events" after `clear()` and before `sendKeys` inside `setValue` implementation. No need to fire events twice, we need to do that just once at the end of "atomic user operation", i.e. at the end of `setValue()`...

Potential implementation could be to change "clearing" implementation to be based on js instead of `WebElement#clear()`. We also can put it under Configuration.clearByJsInsideSetValue or Configuration.fireUnfocusingEventsAfterClearInSetValue or Configuration.someBetterName :)

P.S. by the way `Configuration.fastSetValue` can be used as workaround so far... Yet I believe that just existance of `fastSetValue` is not enough... For me having events being fired twice inside setValue just does not make sense from user perspective...

**Comment by rosolko** [comment]
@yashaka Proposed potential implementation with clear using is also not an "atomic user operations".  Since it's a selenium implementation change it for some we can also go with this because `setValue` method assume such behavior. And yes `fastSetValue` correctly tune clear flow. For now I don't see real problem and necessity of disable unfocus step built in a `clear` selenium method.

**Comment by yashaka** [comment]
under atomic I mean "atomic from user perspective". Such user operation has one "firing events" step - in the end. But now there are two of them, which I believe does not make sense in context of "setValue" user action.

**Comment by rosolko** [comment]
Ok. In such perspective we should perform "select all" and type a new string then. Not a clean + type.

**Comment by yashaka** [comment]
Yes, you are right... Makes sense... Another question - how to do it in a
portable to mobile devices way o_O ... (javascript version would also be
not portable...)

**Comment by asolntsev** [comment]
@yashaka @rosolko I agree, `$.setValue()` should trigger events only once. 
In case of `fastSetValue=true` it works correctly, so we have the problem only in case of ``fastSetValue=false`. 

Yes, `Ctrl+A+SendKeys(newText)` should solve the problem. Can somebody test it in mobile?

**Comment by vinogradoff** [comment]
There is probably another option worth discussing: always do clear through JS. Does it work for mobile?

	  //element.clear();
      setValueByJs(driver,element,"");
      element.sendKeys(text);
      
This code works on desktop for react todomvc.

**Comment by vinogradoff** [comment]
see also #1497

**Comment by asolntsev** [comment]
hopefully implemented in https://github.com/selenide/selenide/pull/1787
